src/ui/components/shared/MaterialIcon.tsx

```diff
@@ -18,6 +18,22 @@ type MaterialIconProps = React.HTMLProps<HTMLDivElement> & {
   iconSize?: keyof typeof SIZE_STYLES;
 };
 
+let isChecking = false;
```

Comment:
Can inline inside the effect by creating a ref object (`useRef`). The effect could then be reused in multiple places (not that we want, but still better encapsulation).